### ISSUE TITLE
DEPR: sort=True in DataFrame.unstack and Series.unstack

### ISSUE THREAD

**Issue by rhshadrach** [pandas-dev/pandas#53915]
The `sort` argument was added in #53298 because of #15105. Now that users can get the unsorted result, they can use `sort=False` and call `.sort_index` after unstacking if they wish to have sorted columns.

**Comment by jorisvandenbossche** [comment]
The keyword was only added in main (i.e. not yet released), so there is no need to deprecate it first if we still want to change it? 

Or it's the default of `True` that you want to change to False eventually?

**Comment by jorisvandenbossche** [comment]
And this is only for `unstack`, right? (now that the `sort` keyword has been removed again from `stack` in https://github.com/pandas-dev/pandas/pull/53825)

**Comment by rhshadrach** [comment]
> Or it's the default of `True` that you want to change to False eventually?

Yes - the goal is to make `sort=False` the only behavior. So we would deprecate both not specifying `sort` and specifying `sort=True`. Then in 3.0, `sort` would only be able to be False, and we would deprecate the argument entirely.

**Comment by rhshadrach** [comment]
Clearing the milestone, there are currently issues with sort=False. See #55516.
